apps/source_academy/lib/source_academy/user.ex

```diff
@@ -0,0 +1,48 @@
+defmodule SourceAcademy.User do
+  @moduledoc """
+    The User entity contains basic user data such as name and e-mail.
+    A user can have many authorisations from different providers.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias SourceAcademy.Repo
+
+  @type t :: %__MODULE__{}
+
+  schema "users" do
+    field :name, :string
+    field :email, :string
+    field :is_admin, :boolean
+
+    has_many :authorizations, SourceAcademy.Authorization
+
+    timestamps()
+  end
+
+  @required_fields ~w(name email)a
+  @optional_fields ~w()a
+
+  def build(params) do
+    changeset(%SourceAcademy.User{}, params)
+  end
+
+  def registration_changeset(user, params \\ :empty) do
+    user
+    |> cast(params, ~w(email name)a)
+    |> validate_required(@required_fields)
+    |> validate_format(:email, ~r/.*@.*/)
+  end
+
+  def changeset(user, params \\ :empty) do
+    user
+    |> cast(params, @required_fields ++ @optional_fields)
+    |> validate_required(@required_fields)
+    |> validate_format(:email, ~r/.*@.*/)
+  end
+
+  def make_admin!(user) do
```

Comment:
make_student, make_staff

make_admin shouldn't be allowed I think, too dangerous.
Let's just seed with two admins for now (Martin and Head TA)